strongbox-commons/src/main/java/org/carlspring/strongbox/io/MultipleDigestInputStream.java

```diff
@@ -232,28 +232,25 @@ public void reload()
     public void reposition()
             throws IOException
     {
-        if (byteRanges != null && !byteRanges.isEmpty() && currentByteRangeIndex < byteRanges.size())
+        if (byteRanges != null && !byteRanges.isEmpty() && currentByteRangeIndex < byteRanges.size() && currentByteRangeIndex < byteRanges.size())
```

Comment:
This seems to be a duplication...? The original code seems to have also had it. (There are two checks for `currentByteRangeIndex < byteRanges.size()`).